src/cockpit/channels/metrics.py

```diff
@@ -15,23 +15,147 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import asyncio
+import json
+import sys
+import time
 import logging
+from typing import Any, List, NamedTuple
 
-from ..channel import Channel
+from ..channel import AsyncChannel
+from ..samples import Sampler, SAMPLE_DESCRIPTIONS
+from ..samples import (cockpit_cpu_samples, cockpit_memory_samples, Samples)
 
 logger = logging.getLogger(__name__)
 
 
-class MetricsChannel(Channel):
+class MetricDescription(NamedTuple):
+    name: str
+    units: str
+    semantics: str
+    instanced: List
+    sampler: int
+
+
+class MetricInfo(NamedTuple):
+    desc: MetricDescription
+    derive: str
+    instances: List
+    value: Any
+
+
+class InternalMetricsChannel(AsyncChannel):
     payload = 'metrics1'
+    restrictions = {'source': 'internal'}
+    metrics: List[MetricInfo] = []
```

Comment:
I'd put a newline before this to separate it from the above data.

Reply:
Possibly also use `_` to show they're locals.

And in any case, don't assign `[]` there, please.  This is a dangerous value.